adobe/react-spectrum
# fix(#4382): [Accessibility] TableView: Screen Readers do not work in Firefox

At present, we do not support sections in TableView; the rendered content within the ScrollView is {state.visibleViews[1]}, just the table body content. I think we can fix #4382, by using role="presentation" for the TableRowGroup returned by the renderWrapper method at https://github.com/adobe/react-spectrum/blob/f1d3a8d353c4b5a41a463b08b19d69e5f7e1a1d8/packages/%40react-spectrum/table/src/TableViewBase.tsx#L294, and and then replacing role="presentation" on the ScrollView at https://github.com/adobe/react-spectrum/blob/f1d3a8d353c4b5a41a463b08b19d69e5f7e1a1d8/packages/%40react-spectrum/table/src/TableViewBase.tsx#L620 with role="rowgroup" at https://github.com/adobe/react-spectrum/blob/f1d3a8d353c4b5a41a463b08b19d69e5f7e1a1d8/packages/%40react-spectrum/table/src/TableViewBase.tsx#L638.


Closes #4382

## ✅ Pull Request Checklist:

- [x] Included link to corresponding React Spectrum GitHub Issue #4382.

## 📝 Test Instructions:

1. Open React-Spectrum TableView Documentation page from this PR's build https://reactspectrum.blob.core.windows.net/reactspectrum/a0c4b996b72cdb148377ab4b31a707b1862d0880/docs/react-spectrum/TableView.html
2. Run axe DevTools within the Chrome web inspector.
3. 6 [Scrollable region must have keyboard access](https://dequeuniversity.com/rules/axe/4.9/scrollable-region-focusable?application=AxeChrome) error warnings will be returned. These are known false positives because each scrolling table body with CSS `overflow: auto` and `role="rowgroup"` has `tabIndex={-1}`, yet each table contains focusable content.

Contrast this with the previous behavior:
1. Open React-Spectrum TableView Documentation page from https://react-spectrum.adobe.com/react-spectrum/TableView.html
2. Run axe DevTools within the Chrome web inspector.
3. The 6 [Scrollable region must have keyboard access](https://dequeuniversity.com/rules/axe/4.9/scrollable-region-focusable?application=AxeChrome) known false positives are still there.
4. In addition, 38 [Certain ARIA roles must contain particular children](https://dequeuniversity.com/rules/axe/4.9/aria-required-children?application=AxeChrome) error warnings are returned, because the scrolling table body with CSS `overflow: auto` has both `role="presentation"` and `tabIndex={-1}`. Browsers interpret the scrollable element as an additional generic element in the accessibility tree, which breaks the expected table structure in WAI-ARIA.

## 🧢 Your Project:

Adobe/Accessibility
